packages/field-base/src/label-controller.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { SlotController } from '@vaadin/component-base/src/slot-controller.js';
+
+/**
+ * A controller to manage the label element.
+ */
+export class LabelController extends SlotController {
+  constructor(host, labelId) {
+    super(
+      host,
+      'label',
+      () => document.createElement('label'),
+      (host, node) => {
+        this.label = host.label;
+
+        this.__savedLabelId = labelId;
+
+        // Set ID attribute or use the existing one.
+        this.__updateLabelId(node);
+
+        // Set text content for the default label.
+        this.__applyDefaultLabel(this.label, node);
+
+        this.__observeLabel(node);
+      }
+    );
+  }
+
+  hostConnected() {
+    super.hostConnected();
+
+    const labelNode = this.getSlotChild();
+    if (labelNode !== this.defaultNode) {
+      this.initCustomNode(labelNode);
+    }
```

Comment:
Why is it needed? Isn't it already covered by the `SlotController.observe` method? What do you think about calling `this.__slotObserver.flush()` in `SlotController.hostConnected` instead?

Reply:
How about having `initDefaultNode` as well?

Reply:
Done. Moved this logic to `SlotController`.